### PR TITLE
HDC1080 Module Add

### PR DESCRIPTION
**Pull request by saucompeng**
New Module

- [x] This PR is for the `dev` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/en/*`.

\<Description of and rational behind this PR\>

### CODE REVIEW

File: app/modules/hdc1080.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    platform_i2c_send_start(hdc1080_i2c_id);
+    platform_i2c_send_address(hdc1080_i2c_id, hdc1080_i2c_addr, PLATFORM_I2C_DIRECTION_RECEIVER);
+
+    for (i=0; i<2; i++) {
+	data[i] = platform_i2c_recv_byte(hdc1080_i2c_id, 1);
+    }
+
+    platform_i2c_send_stop(hdc1080_i2c_id);
+
+    temp = ((data[0]*256+data[1])/pow(2,16))*165-40;
```
Comment (pjsg) [inline app/modules/hdc1080.c]: Does this work in the integer build? In the float build it seems a pity that you discard the fractional component of the temperature.

File: docs/en/modules/hdc1080.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+This module provides access to the [HDC1080](http://www.ti.com/product/HDC1080) . Low Power, High Accuracy Digital Humidity Sensor with Temperature Sensor.
+
+## hdc1080.readTemperature()
+Samples the sensor and returns temperature value.
+
+#### Syntax
+`hdc1080.readTemperature()`
+
+#### Returns
+Temperature data (integers)
```
Comment (pjsg) [inline docs/en/modules/hdc1080.md]: In what units?

File: docs/en/modules/hdc1080.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+```
+
+## hdc1080.readHumidity()
+Samples the sensor and returns temperature value.
+
+#### Syntax
+`hdc1080.readHumidity()`
+
+#### Returns
+Humidity data (integers)
```
Comment (pjsg) [inline docs/en/modules/hdc1080.md]: I guess that this is a percentage from 0 - 100??

File: app/modules/hdc1080.c [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    uint8_t data[2];
+    int temp;
+    int i;
+
+    platform_i2c_send_start(hdc1080_i2c_id);
+    platform_i2c_send_address(hdc1080_i2c_id, hdc1080_i2c_addr, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+    platform_i2c_send_byte(hdc1080_i2c_id, HDC1080_TEMPERATURE_REGISTER);
+    platform_i2c_send_stop(hdc1080_i2c_id);
+
+    os_delay_us(20000);
```
Comment (pjsg) [inline app/modules/hdc1080.c]: This seems long. The max time taken for the conversion is < 7ms for 14 bit accuracy. I don't see why you are requesting 14 bit accuracy when you only return an integer temperature. If you requested 11, then the conversion time is under 4ms.....

Similarly for the humidity....

File: docs/en/modules/hdc1080.md (hunk 3 above)
Comment (saucompeng) [inline docs/en/modules/hdc1080.md]: This is percentage, I will fix

File: app/modules/hdc1080.c (hunk 4 above)
Comment (saucompeng) [inline app/modules/hdc1080.c]: True, I will change it.

File: app/modules/hdc1080.c (hunk 1 above)
Comment (saucompeng) [inline app/modules/hdc1080.c]: True, I change it to float.

File: docs/en/modules/hdc1080.md (hunk 2 above)
Comment (saucompeng) [inline docs/en/modules/hdc1080.md]: Sorry :) in Centigrade

File: app/modules/hdc1080.c (hunk 4 above)
Comment (saucompeng) [inline app/modules/hdc1080.c]: @pjsg  I'm new  at github, Which branch I should make pull request? Master or dev ?

Comment (pjsg) [inline app/modules/hdc1080.c]: You are doing it right -- the PR should be to nodemcu:dev  -- we then merge this to master every 2-3 months when things look good.

File: app/modules/hdc1080.c (hunk 1 above)
Comment (pjsg) [inline app/modules/hdc1080.c]: You need to be careful -- there are two builds of the nodemcu firmware -- one is integer only and the other includes floats. This makes this sort of thing complicated!

Comment (saucompeng) [inline app/modules/hdc1080.c]: https://github.com/nodemcu/nodemcu-firmware/blob/dev/docs/en/modules/dht.md

> If using float firmware then `temp` and `humi` are floating point numbers. On an integer firmware, the final values have to be concatenated from `temp` and `temp_dec` / `humi` and `hum_dec`.

Is it true method  ?

**Pull request closed without merging**